src/styles/main.scss

```diff
@@ -1,3 +1,77 @@
+$row-width: 100px;
+$gap: 1px;
+$days: (
+  "mon": 0,
+  "tue": 1,
+  "wed": 2,
+  "thu": 3,
+  "fri": 4,
+  "sat": 5,
+  "sun": 6,
+);
+
+@function calendarWidth($boxWidth, $gap) {
+  @return (($boxWidth * 7) + ($gap * 6));
+}
+
+@mixin boxSize($value) {
+  height: #{$value};
+  width: #{$value};
+}
+
 body {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  width: calendarWidth($row-width, $gap);
+
+  @each $day, $pos in $days {
+    &--start-day-#{$day} :first-child {
+      margin-left: ($row-width + $gap) * $pos;
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} :nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    position: relative;
+    box-sizing: border-box;
+    border: 1px solid black;
+    @include boxSize($row-width);
+
+    transition: 500ms;
```

Comment:
Specify to what properties the transition should apply

Reply:
I didn’t specify it because I want it to apply to every property

Reply:
@Fluebubble If you set transitions for all properties, you capture those that you will not use, which will once again overload your styles.
This won't be noticeable on simple tasks, but on large ones it will be very noticeable, so specifying specific properties is good practice for specifying styles